src/lib/ctable.lua

```diff
@@ -217,6 +217,30 @@ function CTable:insert(hash, key, value, updates_allowed)
    return index
 end
 
+-- Choose a random index between the start of the table and its occupancy_hi
+-- value. This guarantees that there will be an entry to eject, because this
+-- is only called when the table is 'full'.
+local function random_eject(ctab)
+   local eject_index = math.random(0, ctab.occupancy_hi - 1)
```

Comment:
Alternately you could choose a random hash within [0, HASH_MAX).  Then map to an index using the routine provided or something, starting over at 0 if none was found.